designs/helm-chart-autogenerate-plugin.md

```diff
@@ -0,0 +1,271 @@
+
+Lastly, when new API(s) (CRDs) are introduced, they must be added to the Helm Chart as well. As a result, the Makefile targets make generate and make manifest should be modified when the proposed plugin is in use to ensure that CRDs are accurately copied and pasted into the Helm Chart's respective directory.
+
+To allow users to customize their Helm Charts, we should avoid overwriting files within the Helm Chart, with the exception of CRDs/webhooks, as changes made to these must be mirrored in the Helm Chart. However, what is scaffolded by the init/edit should remain primarily unchanged. We would only uncomment the options as described above.
+
+## Motivation
+
+Kubebuilder users currently face the challenges of lacking an officially supported distribution mechanism for Helm Chart. They seek to:
+
+- Harness the power of Helm Chart as a package manager for the Operator, enabling seamless adaptation to diverse deployment environments.
+- Take advantage of Helm's dependency management capabilities to simplify the installation process of Operator dependencies, such as cert-manager.
+- Seamlessly integrate with Helm's ecosystem, including FluxCD, to efficiently manage the Operator.
+
+However, Kubebuilder does not offer any utilities to facilitate the distribution of projects currently. One could clone the entire project and utilize the Makefile targets to apply the solution on clusters. Consequently, this proposal aims to introduce a method that allows Kubebuilder users to easily distribute their projects through Helm Charts, a strategy that many similar many well-known operators contains the distribution of helm charts:
+
+- [mongodb](https://artifacthub.io/packages/helm/mongodb-helm-charts/community-operator
+)
+- [cert-manager](https://cert-manager.io/v1.6-docs/installation/helm/#1-add-the-jetstack-helm-repository)
+- [prometheus](https://bitnami.com/stack/prometheus-operator/helm)
+- [aws-load-balancer-controller](https://github.com/kubernetes-sigs/aws-load-balancer-controller/tree/main/helm/aws-load-balancer-controller)
+
+
+**NOTE** For further context see the [discussion topic](https://github.com/kubernetes-sigs/kubebuilder/discussions/3074) 
+
+### Goals
+
+- Allow Kubebuilder users distribute their projects using Helm easily.
+- Make the best effort to preserve any customizations made to the Helm Charts by the users.
+
+### Non-Goals
+
+- Converting any Kustomize configuration to Helm Charts like [helmify](https://github.com/arttor/helmify) does.
+- Support the deprecated plugins. This option should be supported from go/v4 and kustomize/v2
+- Introduce support for Helm in addition to Kustomize, or replace Kustomize with Helm entirely, similar to the approach taken by Operator-SDK, thereby allowing users to utilize Helm Charts to build their Operators.
+
```

Comment:
```suggestion
- Attend standard practices that deviate from HelmCharts layout, definition, or conventions to workaround its limitations
```

We need to add more one here.
Otherwise, we will end up in the business to start to discuss how company/team A does and how company/team B does when we cannot attend to everybody. So, we need to stick with Helm definitions and official docs to generate the HelmChart

Reply:
Agree.